datafusion/common/src/config.rs

```diff
@@ -1249,6 +1246,15 @@ impl TableOptions {
         clone
     }
 
+    /// Sets the file format for the table.
+    ///
+    /// # Parameters
+    ///
+    /// * `format`: The file format to use (e.g., CSV, Parquet).
+    pub fn set_file_format(&mut self, format: ConfigFileType) {
```

Comment:
the parameter is called "format" but the struct is `ConfigFileType`

Maybe we should call this `set_config_type`? Or maybe it could just pass in a `Arc<dyn FIleFormat>~?